### DISCUSSION TITLE
Yet another permissions issue

### DISCUSSION THREAD

**Discussion by lowlyIntern** [jlesage/docker-handbrake#316] [Q&A]
I'm new to Docker and banging my head against the wall -- thanks in advance for any help you can give.

I can get Handbreak running and access it via the browser, but am unable to open any of my storage directories, and get an error stating "Could not read the contents of storage. Error opening directory '/storage': Permission denied."

When I check the logs, they also show a permissions issue, with it reporting it can't access the "watch" folder.

I've tried redefining the USER_ID and GROUP_ID from my main admin account, another user, as well as 1000/1000 for each. I've checked and double-checked my syntax for defining the volumes in question.

This is running on Synology's Container Manager, DSM 7.2.1-69057

I'm trying to use a Docker Compose file, and based mine off of the provided [starter file](https://github.com/jlesage/docker-handbrake#docker-compose-file).

I've also tried doing the same troubleshooting steps with multiple USER_IDs and GROUP_IDs via the Container tab where one can simply input values into the GUI to skip the command-line to spin up a container, all with no luck.

Here's my current Docker Compose file:

> version: '3'
services:
  handbrake:
    image: jlesage/handbrake
    devices: 
      - /dev/dri:/dev/dri
    environment:
      - USER_ID = "1029"
      - GROUP_ID = "100"
    ports:
      - "5800:5800"
    volumes:
      - "/volume1/docker/handbrake:/config:rw"
      - "/volume1/media:/storage:ro"
      - "/volume1/media/watch:/watch:rw"
      - "/volume1/media/output:/output:rw"
      - "/volume1/media/#recycle:/trash:rw"

volume1 is my DiskStation's only volume.
volume1/docker/ is where I have all my various container's configuration files, etc.
volume1/media/ is where all my media lives, with the appropriate /watch, /output, and /#recycle subdirectories.

I imagine there is something rather straight-forward that I'm missing, but would appreciate any guidance. Thanks for your time and attention.

**Comment by jlesage** [comment]
I would first check who owns `/volume1/media`:
```
ls -ld /volume1/media
```

**Reply by lowlyIntern** [reply]
Thanks for the suggestion!

When I ssh in and check the permissions, it looks like it's owned by root -- so I clearly don't want to run as that. That directory is one of the Synology set-up "Shared" directories, so I suppose that makes sense. Interestingly, the subdirectories are owned by my main account, except for the "output" directory. I deleted and remade that directory in File Station, and verified it had the permissions from myMainAccount; I then double-checked that I have the right USER_ID and GROUP_ID set, then cleaned and rebuilt -- same problem, and /volume1/media/output's owner was changed to USER_ID = 1000, GROUP_ID = 1000.

> ls -ld /volume1/media
> drwxrwxrwx+ 1 root root 166 Jan  8 18:50 /volume1/media

> ls -ld /volume1/media/movies/
> 1 myMainAccount users 18906 Nov 27 22:42 /volume1/media/movies/

> ls -ld /volume1/media/tv/
> drwxrwxrwx+ 1 myMainAccount users 4678 Nov 27 22:11 /volume1/media/tv/

> ls -ld /volume1/media/output/
> drwxrwxrwx+ 1 1000 1000 0 Jan  7 10:54 /volume1/media/output/

> ls -ld /volume1/media/watch/
> drwxrwxrwx+ 1 myMainAccount users 0 Jan  7 10:54 /volume1/media/watch/
